src/Compilers/CSharp/Portable/Compilation/InitializerSemanticModel.cs

```diff
@@ -181,20 +181,22 @@ private BoundNode BindEqualsValue(Binder binder, EqualsValueClauseSyntax equalsV
             {
                 case SymbolKind.Field:
                     {
-                        var enumField = this.MemberSymbol as SourceEnumConstantSymbol;
+                        var fieldSymbol = (FieldSymbol)this.MemberSymbol;
+                        var enumField = fieldSymbol as SourceEnumConstantSymbol;
                         if ((object)enumField != null)
                         {
                             return binder.BindEnumConstantInitializer(enumField, equalsValue.Value, diagnostics);
                         }
 
-                        var fieldType = ((FieldSymbol)this.MemberSymbol).GetFieldType(binder.FieldsBeingBound);
-                        return binder.BindVariableOrAutoPropInitializer(equalsValue, fieldType, diagnostics);
+                        var fieldType = (fieldSymbol).GetFieldType(binder.FieldsBeingBound);
+                        return new BoundFieldInitializer(equalsValue, fieldSymbol, binder.BindVariableOrAutoPropInitializer(equalsValue, fieldType, diagnostics));
                     }
 
                 case SymbolKind.Property:
                     {
-                        var propertyType = ((PropertySymbol)this.MemberSymbol).Type;
-                        return binder.BindVariableOrAutoPropInitializer(equalsValue, propertyType, diagnostics);
+                        var propertySymbol = (PropertySymbol)this.MemberSymbol;
+                        var propertyType = propertySymbol.Type;
+                        return new BoundPropertyInitializer(equalsValue, propertySymbol, binder.BindVariableOrAutoPropInitializer(equalsValue, propertyType, diagnostics));
```

Comment:
Rather than using BoundFieldInitializer/BoundPropertyInitializer in here, I would introduce a new bound node, BoundEqualsValue derived from BoundNode with only one property - Value (we don't really need the target symbol), and would change signature of this function to return BoundEqualsValue node. This way all initializers will be handled uniformly here. For example, why are we not changing ```case SymbolKind.Parameter```? Is just because we don't have a unit-test for IOperation that goes through it? Perhaps we should add one? It feels like we should have unit-tests for IOperation covering every single branch in this function.